src/engraving/libmscore/harppedaldiagram.cpp

```diff
@@ -0,0 +1,342 @@
+/*
+ * SPDX-License-Identifier: GPL-3.0-only
+ * MuseScore-CLA-applies
+ *
+ * MuseScore
+ * Music Composition & Notation
+ *
+ * Copyright (C) 2022 MuseScore BVBA and others
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include "harppedaldiagram.h"
+
+#include "score.h"
+#include "segment.h"
+#include "types/typesconv.h"
+#include "rw/xml.h"
+
+using namespace mu;
+using namespace mu::engraving;
+
+// Position of separator character in diagram
+static const int SEPARATOR_IDX = 3;
+
+static const String harpStringTypeToString(HarpStringType type)
+{
+    switch (type) {
+    case HarpStringType::A:
+        return String("A");
+    case HarpStringType::B:
+        return String("B");
+    case HarpStringType::C:
+        return String("C");
+    case HarpStringType::D:
+        return String("D");
+    case HarpStringType::E:
+        return String("E");
+    case HarpStringType::F:
+        return String("F");
+    case HarpStringType::G:
+        return String("G");
+    }
+    return String("");
```

Comment:
Better just `String()` instead of `String("")`.

Also, those `String` literals should be prefixed with `u`. So: `String(u"A")`. 

Otherwise, you are converting from `const char*` to QString and then from QString to String. That's undesired because we want that the engraving module doesn't depend on Qt.